### PR TITLE
[#3669] add a script to fix the translations ids

### PR DESCRIPTION
**Pull request by valllllll2000**
#### Before the PR (what is the issue or what needed to be done)
Some translations are missing surveyId and surveyGroupId and with the new logic #3669  they would not be copied
#### The solution
Create a script to fix the data
#### Screenshots (if appropriate)
NA
#### Reviewer Checklist
* [ ] Added an explanation about the work done
* [ ] Connected the PR and the issue on Zenhub
* [ ] Added a test plan to the issue
* [ ] Updated the copyright header (when relevant)
* [ ] Formatted the code
* [ ] Added a documentation (if relevant)
* [ ] Added some unit tests (if relevant)

### CODE REVIEW

File: scripts/data/src/org/akvo/gae/remoteapi/FixOldTranslations.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        translations.addAll(entities);
+        System.out.println("Found " + translations.size() + " broken translations");
+        return translations;
+    }
+
+    private Entity fetchGroup(DatastoreService ds, Long groupId) {
+        if (groupId == null) {
+            return null;
+        }
+        final Query.Filter fs = new Query.FilterPredicate(Entity.KEY_RESERVED_PROPERTY,
```
Comment (iperdomo) [inline scripts/data/src/org/akvo/gae/remoteapi/FixOldTranslations.java:114]: If you know the `id` of the entity you want, you can just do `.get()`, you don't need to *query* for it:
You need to convert the `Long` value into ` Key`:
https://cloud.google.com/appengine/docs/standard/java/datastore/entities#Retrieving_an_entity

You can get a `Key` passing the `Long` value and the `Kind` name:
https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/KeyFactory.html#createkey_2

File: scripts/data/src/org/akvo/gae/remoteapi/FixOldTranslations.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            return t;
+        }
+        return null;
+    }
+
+    private List<Entity> fetchBrokenTranslations(DatastoreService ds) {
+        List<Entity> translations = new ArrayList<>();
+        final Query q = new Query("Translation");
+        final PreparedQuery pq = ds.prepare(q);
+        final Query.Filter filterNullSurveyId = new Query.FilterPredicate("surveyId", Query.FilterOperator.EQUAL, null);
```
Comment (iperdomo) [inline scripts/data/src/org/akvo/gae/remoteapi/FixOldTranslations.java:101]: I wonder if this query is *"correct"*. In the Datastore a *missing* property gets exposed as a `null` value, when you have already the object in memory, but you can't query for entities with a missing property. See: https://cloud.google.com/appengine/docs/standard/java/datastore/query-restrictions#entities_lacking_a_property_named_in_the_query_are_ignored

If we're sure that the value **is** `null` then just ignore this comment.

File: scripts/data/src/org/akvo/gae/remoteapi/FixOldTranslations.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            }
+        }
+        return null;
+    }
+
+    private Entity fixQuestionOptionTranslation(Entity t, final DatastoreService ds) {
+        System.out.println("Fixing QuestionOption translation");
+        Long parentId = (Long) t.getProperty("parentId");
+        if (parentId != null) {
+            final Query.Filter fs = new Query.FilterPredicate(Entity.KEY_RESERVED_PROPERTY,
```
Comment (iperdomo) [inline scripts/data/src/org/akvo/gae/remoteapi/FixOldTranslations.java:73]: Use `ds.get()` if we know the `id` of the entity we want

**Review by iperdomo — CHANGES_REQUESTED** [review]
I've added a couple of comments:
* Use `ds.get(<Key>)` for entities we know the `id`
* We can't search for entities with missing properties, so querying for `null` can be misleading

File: scripts/data/src/org/akvo/gae/remoteapi/FixOldTranslations.java (hunk 2 above)
Comment (valllllll2000) [reply scripts/data/src/org/akvo/gae/remoteapi/FixOldTranslations.java:101]: yes I am sure this works, I have tested it and Translations do get found with these missing properties and loads of them

**Comment by valllllll2000** [comment]
Fixed: getting entities by Key, As for null, I am sure this is the value per default as I have tried it on several dev instances and for sandbox for example it found 60 such entities which now have been fixed.

**Review by iperdomo — APPROVED** [review]

**Pull request merged**